lib/spear.ex

```diff
@@ -413,6 +415,9 @@ defmodule Spear do
            messages,
            Keyword.take(opts, [:credentials, :timeout])
          ) do
+      {:ok, Streams.append_resp(result: {:success, _} = response)} when raw? == true ->
+        {:ok, response}
```

Comment:
I believe we should return the whole `Streams.append_resp/1` record to be consistent with other `raw?: true` behavior

```suggestion
      {:ok, response} when raw? == true ->
        {:ok, response}
```

Reply:
Right I was thinking that when `raw?` is true then we should return the whole `Streams.append_resp()` even when the result isn't `{:success, _}`, so the `:wrong_expected_version` would be returned raw as well

Reply:
ok. so would you prefer `append/3`:

1. return `{:ok, raw_response}` in all cases, whether or not the raw response is success or wrong expected version.

2. Or, return `{:ok, raw_response}` when success;
and `{:error, raw_response}` when `:wrong_expected_version` is received?

In both cases the raw response is returned, but the status `:ok` vs `:error` will be different.

Reply:
Ah I see what you're saying now, I missed that in the last message and the diff. I was originally thinking that both would return `{:ok, append_response}`. It's unfortunately inconsistent with how we behave when it's `raw?: false` but the idea of the `:raw?` option is to not look at the response type at all and leave that to the caller

Reply:
> but the idea of the :raw? option is to not look at the response type at all and leave that to the caller

roger that. I'll make the change accordingly, per your suggestion

```
      {:ok, response} when raw? == true ->
        {:ok, response}
```